docker/nginx-template.conf

```diff
@@ -0,0 +1,37 @@
+# Do not update nginx.conf manually
+# The config file would be generated when docker-compose starts.
+# Use generate-nginx-conf.py to generate the file manully.
+events {{
+
+}}
+
+http {{
+    map $http_upgrade $connection_upgrade {{
+        default upgrade;
+        '' close;
+    }}
+
+    upstream gateways {{
+        ip_hash;
+        {gateways}
+    }}
+
+    server {{
```

Comment:
Optional: For ease of deployment, I would recommend having only 1 virtual host, that way we can make it the default, and possibly do away with cross origin entirely. The advantage is that if someone wants to spin up an instance of hitcon-online, there would be no need to setup multiple DNS before doing docker compose up. We do have only 1 virtual host by having different location point to different upstream. i.e. location /static and location /client.html goes to asset service.